app.rb

```diff
@@ -0,0 +1,137 @@
+require './book'
+require './person'
+require './rental'
+require './student'
+require './teacher'
+require './classroom'
+
+class App
+  def initialize(menu)
+    @menu = menu
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def list_all_books
+    puts 'There are no books yet! Kindly add books.' if @books.empty?
+
+    @books.each { |book| puts "Title: #{book.title}, Author: #{book.author}" }
+    sleep 0.75
+    @menu.display_menu
+  end
+
+  def list_all_people
+    puts 'There are no people yet! Kindly add a student or teacher.' if @people.empty?
+    @people.map { |person| puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}" }
+    sleep 0.75
+    @menu.display_menu
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or teacher (2) [Input a number]: '
+    option = gets.chomp
+
+    case option
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'Invalid input. kindly type 1 or 2'
+    end
+    @menu.display_menu
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    print 'Name: '
+    name = gets.chomp
+
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp.downcase
+
+    print 'Classroom: '
+    classroom = gets.chomp
+
+    student = Student.new(age, classroom, name, parent_permission)
+    @people.push(student)
+
+    puts 'Student created successfully'
+    sleep 0.75
+  end
+
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    print 'Name: '
+    name = gets.chomp
+
+    print 'Specialization: '
+    specialization = gets.chomp
+
+    teacher = Teacher.new(age, specialization, name)
+    @people << teacher
+
+    puts 'Teacher created successfully'
+    sleep 0.75
+  end
```

Comment:
- [ ] I tried to add a new person, teacher but I got an error after I input the specialization option. See the error below -

**Error**
![image](https://user-images.githubusercontent.com/84156714/216773809-111f20a6-53e9-4088-a591-81023bf26f4b.png)

Please fix this.

Reply:
Okay, thank you for the observation. Error fixed.